src/connectors/pb.rs

```diff
@@ -0,0 +1,323 @@
+// Copyright 2020-2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::errors::Result;
+use simd_json::StaticNode;
+use tremor_value::Value;
+
+pub(crate) fn maybe_string_to_pb(data: Option<&Value<'_>>) -> Result<String> {
+    if let Some(Value::String(s)) = data {
+        Ok(s.to_string())
+    } else if data.is_none() {
+        Ok("".to_string())
+    } else {
+        Err("Expected an json string to convert to pb string".into())
+    }
+}
+
+pub(crate) fn maybe_int_to_pbu64(data: Option<&Value<'_>>) -> Result<u64> {
+    if let Some(&Value::Static(StaticNode::U64(i))) = data {
+        Ok(i)
+    } else if let Some(&Value::Static(StaticNode::I64(i))) = data {
+        #[allow(clippy::clippy::cast_sign_loss)]
+        Ok(i as u64)
+    } else {
+        Err("Expected an json u64 to convert to pb u64".into())
+    }
+}
+
+pub(crate) fn maybe_int_to_pbi32(data: Option<&Value<'_>>) -> Result<i32> {
+    if let Some(&Value::Static(StaticNode::I64(i))) = data {
+        #[allow(clippy::clippy::cast_sign_loss)]
+        #[allow(clippy::cast_possible_truncation)]
+        Ok(i as i32)
+    } else {
+        Err("Expected an json i64 to convert to pb i32".into())
+    }
+}
+
+pub(crate) fn maybe_int_to_pbi64(data: Option<&Value<'_>>) -> Result<i64> {
+    if let Some(&Value::Static(StaticNode::I64(i))) = data {
+        Ok(i)
+    } else {
+        Err("Expected an json i64 to convert to pb i64".into())
+    }
+}
+
+pub(crate) fn maybe_int_to_pbu32(data: Option<&Value<'_>>) -> Result<u32> {
+    if let Some(&Value::Static(StaticNode::U64(i))) = data {
```

Comment:
`as_u32`

Reply:
```rust
if let Some(i) = data.as_u32() {
...
}
```

Reply:
Oh these `as_*` conveniences are nice - next commit they'll replace the current form!

Reply:
Done

Reply:
Marking as resolved - we'll pick up `as_u64` with the simd-json `v0.4` release for the last non `as_*`